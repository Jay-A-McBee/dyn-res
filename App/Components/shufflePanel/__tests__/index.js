import React from 'react';
import "babel-polyfill";
import {render, fireEvent, cleanup, waitForElement} from 'react-testing-library'
import {
    WorkDescription,
    Experience              
} from '../index';

jest.mock('../../../Assets/pics/adminScreen.png', () => '');
jest.mock('../../../Assets/pics/spltscreen.png', () => '');
jest.mock('../../../Assets/pics/fairshareShell.png', () => '');
jest.mock('../../../Assets/pics/journeymenShell.png', () => '');
jest.mock('../../../Assets/pics/sentimentalistShell.png', () => '');
jest.mock('@fortawesome/react-fontawesome', () => ({
    FontAwesomeIcon: () => (<div></div>)
}));

const description = {
    a: 'bar',
    b: 'baz',
    c: 'foo'
};

afterEach(cleanup);

describe('<WorkDescription />', () => {
    test('accepts title, description, dates, href, selected props', () => {
        const {getByTestId} = render(
            <WorkDescription
                title={'Foo'}
                description={description}
                dates={'March 2000 - Now'}
                selected={'clicktripz'}
                href={'https://foo.bar'}
            />
        );

        expect(getByTestId('title').textContent).toBe('Foo clicktripz');
        expect(getByTestId('dates').textContent).toBe('March 2000 - Now');
        expect(getByTestId('workLink').textContent).toBe('clicktripz');
        expect(getByTestId('a').textContent).toBe('bar');
        expect(getByTestId('b').textContent).toBe('baz');
        expect(getByTestId('c').textContent).toBe('foo');
    });

    test('will conditionally render a modal when selected has carousel children els', () => {
        const {getByTestId} = render(
            <WorkDescription
                title={'Foo'}
                description={description}
                dates={'March 2000 - Now'}
                selected={'HackReactor'}
                href={'https://foo.bar'}
            />
        );
        expect(getByTestId('modalTest')).toBeTruthy();
    });
});

describe('<Experience />', async () => {

    test('renders clickable tabs and associated description', async () => {
        const {
            getByTestId, 
            container, 
            getByText
        } = render(<Experience />);

        const [
            tab1,
            tab2,
            tab3
        ] = [
            getByTestId('tab1'),
            getByTestId('tab2'),
            getByTestId('tab3')
        ];

        const fullDescription = getByTestId('fullDescription');

        fireEvent.click(tab1);
        expect(getByTestId('tabs')).toMatchSnapshot();
        expect(fullDescription).toMatchSnapshot();
        fireEvent.click(tab2);
        expect(getByTestId('tabs')).toMatchSnapshot();
        expect(fullDescription).toMatchSnapshot();
        fireEvent.click(tab3);
        expect(getByTestId('tabs')).toMatchSnapshot();
        expect(fullDescription).toMatchSnapshot();
    });
});
