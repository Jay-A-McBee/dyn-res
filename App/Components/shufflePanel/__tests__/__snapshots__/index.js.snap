// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Experience /> renders clickable tabs and associated description 1`] = `
<div
  class="animate sc-eHgmQL ILyXW"
  data-testid="tabs"
>
  <div
    class="animate sc-cvbbAY jnuCuz"
    data-testid="tab1"
    name="ClickTripz"
  >
    ClickTripz
  </div>
  <div
    class="animate sc-cvbbAY mBXEX"
    data-testid="tab2"
    name="SPLT"
  >
    SPLT
  </div>
  <div
    class="animate sc-cvbbAY mBXEX"
    data-testid="tab3"
    name="HackReactor"
  >
    HackReactor
  </div>
  <div
    class="sc-jWBwVP ZEGqK"
    offset="0"
  />
</div>
`;

exports[`<Experience /> renders clickable tabs and associated description 2`] = `
<div
  class="sc-htoDjs sc-iRbamj gqGVyA"
  data-testid="fullDescription"
>
  <p
    class="animate sc-brqgnP edDJug"
    data-testid="title"
  >
    Software Engineer @ 
    <a
      class="sc-jAaTju jOCrVx"
      data-testid="workLink"
      href="https://www.clicktripz.com"
      target="_blank"
    >
      ClickTripz
    </a>
  </p>
  <p
    class="animate sc-cMljjf gvzVKk"
    data-testid="dates"
  >
    March 2017 - Present
  </p>
  <br />
  <div
    class="sc-gZMcBi animate sc-jDwBTQ DOFNs"
  >
    <div />
    <p
      class="sc-gPEVay hWgLEq"
      data-testid="a"
    >
      Write efficient and unobtrusive JS that extends CT propietary ad-tech framework (CTI)
    </p>
  </div>
  <div
    class="sc-gZMcBi animate sc-jDwBTQ DOFNs"
  >
    <div />
    <p
      class="sc-gPEVay hWgLEq"
      data-testid="b"
    >
      Work with S3 data storage and Riot.js to build out admin dashboard analytics tooling
    </p>
  </div>
  <div
    class="sc-gZMcBi animate sc-jDwBTQ DOFNs"
  >
    <div />
    <p
      class="sc-gPEVay hWgLEq"
      data-testid="c"
    >
      Provide direct support to client tech teams integrating CT software on high-traffic sites (~50k-100k uniques/week)
    </p>
  </div>
</div>
`;

exports[`<Experience /> renders clickable tabs and associated description 3`] = `
<div
  class="animate sc-eHgmQL ILyXW"
  data-testid="tabs"
>
  <div
    class="animate sc-cvbbAY mBXEX"
    data-testid="tab1"
    name="ClickTripz"
  >
    ClickTripz
  </div>
  <div
    class="animate sc-cvbbAY jnuCuz"
    data-testid="tab2"
    name="SPLT"
  >
    SPLT
  </div>
  <div
    class="animate sc-cvbbAY mBXEX"
    data-testid="tab3"
    name="HackReactor"
  >
    HackReactor
  </div>
  <div
    class="sc-jWBwVP eYMUBu"
    offset="1"
  />
</div>
`;

exports[`<Experience /> renders clickable tabs and associated description 4`] = `
<div
  class="sc-htoDjs sc-iRbamj gqGVyA"
  data-testid="fullDescription"
>
  <p
    class="animate sc-brqgnP edDJug"
    data-testid="title"
  >
    Software Engineer @ 
    <a
      class="sc-jAaTju jOCrVx"
      data-testid="workLink"
      href="https://www.splt.io"
      target="_blank"
    >
      SPLT
    </a>
  </p>
  <p
    class="animate sc-cMljjf gvzVKk"
    data-testid="dates"
  >
    Sept 2016 - April 2017
  </p>
  <br />
  <div
    class="sc-gZMcBi animate sc-jDwBTQ DOFNs"
  >
    <div />
    <p
      class="sc-gPEVay hWgLEq"
      data-testid="a"
    >
      Contributed to multiple products including an enterprise-first carpooling mobile app, an accompanying analytics dashboard, and a non-emergency medical transport dispatch portal
    </p>
  </div>
  <div
    class="sc-gZMcBi animate sc-jDwBTQ DOFNs"
  >
    <div />
    <p
      class="sc-gPEVay hWgLEq"
      data-testid="b"
    >
      Worked extensively with React and React-Native to create a reusable component library
    </p>
  </div>
  <div
    class="sc-gZMcBi animate sc-jDwBTQ DOFNs"
  >
    <div />
    <p
      class="sc-gPEVay hWgLEq"
      data-testid="c"
    >
      Integrated multiple third-party APIs including PayPal, Stripe, Google Geolocate and Edmund's
    </p>
  </div>
  <button
    class="sc-htpNat fzxCZZ"
    data-testid="modalButton"
  >
    View Work
  </button>
  <div
    class="sc-bwzfXH loqWVw"
    id="modal"
  >
    <div
      class="sc-bdVaJa dZNfYL"
      id="modal_body_SPLT"
    >
      <button
        class="sc-bxivhb cwyHox"
        id="close"
      >
        <div />
      </button>
      <div
        class="sc-jzJRlG dVwAfH"
      >
        <div
          class="sc-gZMcBi sc-cSHVUG hKkEvu"
        >
          <div
            class="sc-htoDjs sc-VigVT lnyvjB"
          >
            <button
              class="sc-kgoBCf fEloWE"
            >
              <div />
            </button>
          </div>
          <div
            class="sc-chPdSV eqlpFM"
          >
            <div
              class="sc-gZMcBi sc-kAzzGY cDtotd"
            >
              <div
                class="sc-kGXeez kJpHJI"
              >
                <img
                  class="sc-kpOJdX eyWmfA"
                  src=""
                />
                <div
                  class="sc-htoDjs sc-jKJlTe eXKKwq"
                >
                  <p
                    class="sc-dxgOiQ iWvLPa"
                  >
                    SPLT Dashboard
                  </p>
                  <p
                    class="sc-ckVGcZ ghPshf"
                  >
                    React, Redux
                  </p>
                  <p
                    class="sc-ckVGcZ ghPshf"
                  >
                    Redux-Saga
                  </p>
                  <p
                    class="sc-ckVGcZ ghPshf"
                  >
                    d3
                  </p>
                </div>
              </div>
              <div
                class="sc-kGXeez kJpHJI"
              >
                <img
                  class="sc-kpOJdX eyWmfA"
                  src=""
                />
                <div
                  class="sc-htoDjs sc-jKJlTe eXKKwq"
                >
                  <p
                    class="sc-dxgOiQ iWvLPa"
                  >
                    SPLT Carpooling
                  </p>
                  <p
                    class="sc-ckVGcZ ghPshf"
                  >
                    React Native
                  </p>
                  <p
                    class="sc-ckVGcZ ghPshf"
                  >
                    Redux
                  </p>
                  <p
                    class="sc-ckVGcZ ghPshf"
                  >
                    Redux-Saga
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            class="sc-htoDjs sc-VigVT caOfaf"
          >
            <button
              class="sc-kgoBCf fEloWE"
            >
              <div />
            </button>
          </div>
        </div>
        <div
          class="sc-gqjmRU hswTjB"
        >
          <div
            class="sc-jTzLTM fvIpIj"
            name="0"
          />
          <div
            class="sc-jTzLTM jKkIsg"
            name="1"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Experience /> renders clickable tabs and associated description 5`] = `
<div
  class="animate sc-eHgmQL ILyXW"
  data-testid="tabs"
>
  <div
    class="animate sc-cvbbAY mBXEX"
    data-testid="tab1"
    name="ClickTripz"
  >
    ClickTripz
  </div>
  <div
    class="animate sc-cvbbAY mBXEX"
    data-testid="tab2"
    name="SPLT"
  >
    SPLT
  </div>
  <div
    class="animate sc-cvbbAY jnuCuz"
    data-testid="tab3"
    name="HackReactor"
  >
    HackReactor
  </div>
  <div
    class="sc-jWBwVP fvGVdR"
    offset="2"
  />
</div>
`;

exports[`<Experience /> renders clickable tabs and associated description 6`] = `
<div
  class="sc-htoDjs sc-iRbamj gqGVyA"
  data-testid="fullDescription"
>
  <p
    class="animate sc-brqgnP edDJug"
  >
    Software Engineering Student
  </p>
  <p
    class="animate sc-cMljjf gvzVKk"
    data-testid="dates"
  >
    March 2016 - June 2016
  </p>
  <br />
  <div
    class="sc-gZMcBi animate sc-jDwBTQ DOFNs"
  >
    <div />
    <p
      class="sc-gPEVay hWgLEq"
      data-testid="a"
    >
      Completed a rigourous class schedule combining computer science fundamentals with modern software development best practices
    </p>
  </div>
  <div
    class="sc-gZMcBi animate sc-jDwBTQ DOFNs"
  >
    <div />
    <p
      class="sc-gPEVay hWgLEq"
      data-testid="b"
    >
      Worked with a modern JS stack including frontend and backend frameworks
    </p>
  </div>
  <div
    class="sc-gZMcBi animate sc-jDwBTQ DOFNs"
  >
    <div />
    <p
      class="sc-gPEVay hWgLEq"
      data-testid="c"
    >
      Produced three full-stack JS applications as part of a four person team that included a bill-splitting app, a local musician directory and a sentiment analysis app
    </p>
  </div>
  <button
    class="sc-htpNat fzxCZZ"
    data-testid="modalButton"
  >
    View Work
  </button>
  <div
    class="sc-bwzfXH loqWVw"
    id="modal"
  >
    <div
      class="sc-bdVaJa dZNfYL"
      id="modal_body_HackReactor"
    >
      <button
        class="sc-bxivhb cwyHox"
        id="close"
      >
        <div />
      </button>
      <div
        class="sc-jzJRlG dVwAfH"
      >
        <div
          class="sc-gZMcBi sc-cSHVUG hKkEvu"
        >
          <div
            class="sc-htoDjs sc-VigVT lnyvjB"
          >
            <button
              class="sc-kgoBCf fEloWE"
            >
              <div />
            </button>
          </div>
          <div
            class="sc-chPdSV eqlpFM"
          >
            <div
              class="sc-gZMcBi sc-kAzzGY cDtotd"
            >
              <div
                class="sc-kGXeez kJpHJI"
              >
                <img
                  class="sc-kpOJdX eyWmfA"
                  src=""
                />
                <div
                  class="sc-htoDjs sc-jKJlTe eXKKwq"
                >
                  <p
                    class="sc-dxgOiQ iWvLPa"
                  >
                    Fairshare
                  </p>
                  <p
                    class="sc-ckVGcZ ghPshf"
                  >
                    React/Redux
                  </p>
                  <p
                    class="sc-ckVGcZ ghPshf"
                  >
                    Node.js/Express
                  </p>
                  <p
                    class="sc-ckVGcZ ghPshf"
                  >
                    OAuth 2.0
                  </p>
                  <a
                    class="sc-eNQAEJ ipKFzS"
                    href="https://github.com/AngryPulpGophers/fairshare"
                    target="_blank"
                  >
                    <div />
                  </a>
                </div>
              </div>
              <div
                class="sc-kGXeez kJpHJI"
              >
                <img
                  class="sc-kpOJdX eyWmfA"
                  src=""
                />
                <div
                  class="sc-htoDjs sc-jKJlTe eXKKwq"
                >
                  <p
                    class="sc-dxgOiQ iWvLPa"
                  >
                    Journeymen
                  </p>
                  <p
                    class="sc-ckVGcZ ghPshf"
                  >
                    Node.js/Express
                  </p>
                  <p
                    class="sc-ckVGcZ ghPshf"
                  >
                    PostgreSQL
                  </p>
                  <p
                    class="sc-ckVGcZ ghPshf"
                  >
                    Passport OAuth 2.0
                  </p>
                  <a
                    class="sc-eNQAEJ ipKFzS"
                    href="https://github.com/getJourneymen/Journeymen"
                    target="_blank"
                  >
                    <div />
                  </a>
                </div>
              </div>
              <div
                class="sc-kGXeez kJpHJI"
              >
                <img
                  class="sc-kpOJdX eyWmfA"
                  src=""
                />
                <div
                  class="sc-htoDjs sc-jKJlTe eXKKwq"
                >
                  <p
                    class="sc-dxgOiQ iWvLPa"
                  >
                    Sentimentalist
                  </p>
                  <p
                    class="sc-ckVGcZ ghPshf"
                  >
                    Angular
                  </p>
                  <p
                    class="sc-ckVGcZ ghPshf"
                  >
                    Node.js/Express
                  </p>
                  <p
                    class="sc-ckVGcZ ghPshf"
                  >
                    PostgreSQL
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            class="sc-htoDjs sc-VigVT caOfaf"
          >
            <button
              class="sc-kgoBCf fEloWE"
            >
              <div />
            </button>
          </div>
        </div>
        <div
          class="sc-gqjmRU hswTjB"
        >
          <div
            class="sc-jTzLTM fvIpIj"
            name="0"
          />
          <div
            class="sc-jTzLTM jKkIsg"
            name="1"
          />
          <div
            class="sc-jTzLTM jKkIsg"
            name="2"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
